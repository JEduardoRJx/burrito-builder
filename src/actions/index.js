export const setOrders = orders => ({
  type: 'SET_ORDERS',
  orders
});

export const setNewOrder = newOrder => ({
  type: 'NEW_ORDER',
  newOrder
})