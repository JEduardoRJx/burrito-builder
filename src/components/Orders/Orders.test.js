import React from 'react';
import { shallow } from 'enzyme';
import { Orders, mapStateToProps, mapDispatchToProps } from './Orders'
import { setOrders } from '../../actions';

describe('OrdersContainer', () => {

  describe('Orders component', () => {
    let wrapper;
    const mockOrders = [{
      id: 1,
      ingredients: ["beans", "lettuce", "carnitas", "queso fresco", "jalapeno"],
      name: "Pat"
    }, {
      id: 2,
      ingredients: ["beans", "rice", "carnitas", "jalapeno"],
      name: "SAM"
    }]
    const mockSetOrders = jest.fn();

    beforeEach(() => {
      wrapper = shallow(<Orders 
        orders={mockOrders}
        setOrders={mockSetOrders}
      />);  
    });
    
    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });


  });

  describe('mapStateToProps', () => {
    it('should return orders(array)', () => {
      const orders = [{
        id: 1,
        ingredients: ["beans", "lettuce", "carnitas", "queso fresco", "jalapeno"],
        name: "Pat"
      }, {
        id: 2,
        ingredients: ["beans", "rice", "carnitas", "jalapeno"],
        name: "SAM"
      }]
      const mockState = {
        orders
      }
      const expected = {
        orders
      }
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected);
    });
  });

  describe('mapDispatchToProps', () => {
    it('calls dispatch with a newOrder when setNewOrder is called', () => {
      const mockDispatch = jest.fn();
      const orders = [{
        id: 1,
        ingredients: ["beans", "lettuce", "carnitas", "queso fresco", "jalapeno"],
        name: "Pat"
      }, {
        id: 2,
        ingredients: ["beans", "rice", "carnitas", "jalapeno"],
        name: "SAM"
      }];
      const actionToDispatch = setOrders(orders);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.setOrders(orders);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });

});