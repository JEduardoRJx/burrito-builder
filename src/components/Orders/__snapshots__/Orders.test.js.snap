// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrdersContainer Orders component should match snapshot 1`] = `
<section>
  <div
    className="order"
    key="1"
  >
    <h3>
      Pat
    </h3>
    <ul
      className="ingredient-list"
    >
      <li
        key="beans"
      >
        beans
      </li>
      <li
        key="lettuce"
      >
        lettuce
      </li>
      <li
        key="carnitas"
      >
        carnitas
      </li>
      <li
        key="queso fresco"
      >
        queso fresco
      </li>
      <li
        key="jalapeno"
      >
        jalapeno
      </li>
    </ul>
  </div>
  <div
    className="order"
    key="2"
  >
    <h3>
      SAM
    </h3>
    <ul
      className="ingredient-list"
    >
      <li
        key="beans"
      >
        beans
      </li>
      <li
        key="rice"
      >
        rice
      </li>
      <li
        key="carnitas"
      >
        carnitas
      </li>
      <li
        key="jalapeno"
      >
        jalapeno
      </li>
    </ul>
  </div>
</section>
`;
